src/wscript_build

```diff
@@ -1,17 +1,95 @@
+# Detect a Windows build.
+is_win = 'WIN32' in ' '.join(bld.env.keys())
+
+# Detect MSVC
+is_msvc = bld.env.CXX_NAME == 'msvc'
+
 all_sources = bld.path.ant_glob('*.cpp scene-ideas/*.cc scene-terrain/*.cpp')
 common_sources = [f for f in all_sources if f.name.find('canvas-') == -1 and
                                             f.name.find('android') == -1 and
                                             f.name.find('native-state-') == -1 and
                                             f.name.find('gl-state-') == -1 and
                                             f.name.find('main.cpp') == -1]
-common_uselibs = ['libpng']
+local_uselibs = ['libpng-local', 'zlib-local', 'libjpeg-turbo-local']
+
+common_uselibs = []
 common_defines = ['USE_EXCEPTIONS']
 
+if is_win:
+  common_uselibs += local_uselibs
+else:
+  common_uselibs += ['libpng']
+
 libmatrix_sources = [f for f in bld.path.ant_glob('libmatrix/*.cc')
                      if not f.name.endswith('test.cc')]
 
 common_flavor_sources = ['main.cpp', 'canvas-generic.cpp']
 
+libpng_local_sources = [f for f in bld.path.ant_glob('libpng/*.c')]
+zlib_local_sources = [f for f in bld.path.ant_glob('zlib/*.c')]
+
+# Some c sources in libjpeg-turbo are included like headers. So we cannot use globs.
+libjpeg_local_sources = [
+  'jcapimin',
+  'jcapistd',
+  'jccoefct',
+  'jccolor',
+  'jcdctmgr',
+  'jchuff',
+  'jcinit',
+  'jcmainct',
+  'jcmarker',
+  'jcmaster',
+  'jcomapi',
+  'jcparam',
+  'jcphuff',
+  'jcprepct',
+  'jcsample',
+  'jdapimin',
+  'jdapistd',
+  'jdatadst',
+  'jdatasrc',
+  'jdcoefct',
+  'jdcolor',
+  'jddctmgr',
+  'jdhuff',
+  'jdinput',
+  'jdmainct',
+  'jdmarker',
+  'jdmaster',
+  'jdmerge',
+  'jdphuff',
+  'jdpostct',
+  'jdsample',
+  'jerror',
+  'jfdctflt',
+  'jfdctfst',
+  'jfdctint',
+  'jidctflt',
+  'jidctfst',
+  'jidctint',
+  'jidctred',
+  'jmemmgr',
+  'jmemnobs',
+  'jquant1',
+  'jquant2',
+  'jutils',
+  'jsimd_none',
+]
```

Comment:
Given that we don't expect this list to change often, and hence one file per line won't benefit us much in terms of future diffs, could we format this with more files per line?

Reply:
Done.